File: Codeforces/1618F.cpp
#include<bits/stdc++.h>
#define ll long long
using namespace std;

ll reverse(ll x){
    ll acm = 0;
    while(x){
        acm = 2*acm + x%2;
        x/=2;
    }
    return acm;
}

int main(){
    ll a, b;
    set<ll> todos;
    cin >> a >> b;

    
    queue<ll> q;
    q.push(a);
    todos.insert(a);
    while(q.size()){
        ll x = q.front(); q.pop();
        if(reverse(x)>=0 && todos.find(reverse(x))==todos.end()){
            todos.insert(reverse(x));
            q.push(reverse(x));
        }
        if(2*x+1>=0 && reverse(2*x+1)>=0 && todos.find(reverse(2*x+1))==todos.end()){
            todos.insert(reverse(2*x+1));
            q.push(reverse(2*x+1));
        }
    }

    bool ok = (todos.find(b)!=todos.end());
    cout << ((ok) ? "YES" : "NO") <<"\n";

    return 0;
}

File: Codeforces/1618F.py
def reverse(a):
    acm = 0
    while(a!=0):
        acm = 2*acm + a%2
        a = a//2
    return acm

a, b = [int(x) for x in input().split()]

q = [a]
i = 0
d = {}
d[a] = 1

while(i<len(q) and i<5000):
    x = q[i]
    if(x==b):
        break
    if(reverse(x) not in d):
        d[reverse(x)] = 1
        q.append(reverse(x))

    if((2*x+1 <=b or reverse(2*x+1)<=b) and reverse(2*x+1) not in d):
        d[reverse(2*x+1)] = 1
        q.append(reverse(2*x+1))

    i = i+1
#print(i)
if(b in d):
    print("YES")
else:
    print("NO")

